### PR TITLE
Add type definitions for is-array

### PR DESCRIPTION
**Pull request by pine**
Hello. I added type definitions for is-array module.

- https://github.com/retrofox/is-array
- https://www.npmjs.com/package/is-array

Thank you.

---

Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: is-array/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Type definitions for is-array 1.0
+// Project: https://github.com/retrofox/is-array
+// Definitions by: Pine Mizune <https://github.com/pine>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = isArray;
+declare function isArray(val?: any): boolean;
```
Comment (DanielRosenwasser) [inline is-array/index.d.ts]: Can you make the return type `val is Array<any>`?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: is-array/index.d.ts (hunk 1 above)
Comment (pine) [inline is-array/index.d.ts]: `isArray` can also receive not array types. And it returns boolean always.

https://github.com/retrofox/is-array/blob/master/test/test.html
https://github.com/retrofox/is-array/blob/master/index.js#L14-L29

Comment (DanielRosenwasser) [inline is-array/index.d.ts]: Sure, the return type of `Array.isArray` is the following:

```ts
isArray(arg: any): arg is Array<any>;
```

So it's the same thing. `arg is Array<any>` is a type predicate assertion, which basically says that

1. The function returns a `boolean`
2. If the function returns `true`, then whatever was passed in for `arg` is compatible with `Array<any>`.

Comment (DanielRosenwasser) [inline is-array/index.d.ts]: Also, I do question whether it really makes sense to make it optional. It doesn't seem helpful for anyone.

Comment (aozgaa) [inline is-array/index.d.ts]: Please make the argument required.

Thanks for taking the time to set up the typings :).

Comment (aozgaa) [inline is-array/index.d.ts]: The type of the polyfill should agree with that in `lib.d.ts`:
https://github.com/Microsoft/TypeScript/blob/master/lib/lib.d.ts#L1310

Comment (pine) [inline is-array/index.d.ts]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14344#discussion_r98843794

I did not know that. Thanks!

I'll fix that.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by pine** [comment]
Hello. I've fixed that.
Could you please review it again ? @DanielRosenwasser

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**